### PR TITLE
fix: abseil conflict

### PR DESCRIPTION
**Pull request by MarcusMogg**
fix: abseil conflict in https://github.com/xmake-io/xmake-repo/issues/3705

### CODE REVIEW

File: packages/g/grpc/xmake.lua [hunk 1]
```diff
@@ -35,6 +35,7 @@ package("grpc")
 
     on_load(function (package)
         package:addenv("PATH", "bin")
+        package:add("deps", "abseil", {configs = {cxx_standard = "17"}})
```
Comment (SirLynix) [inline packages/g/grpc/xmake.lua:38]: you can write it in the description scope using `add_deps("abseil", {configs = {cxx_standard = "17"}}})`

File: packages/p/protobuf-cpp/xmake.lua [hunk 2]
```diff
@@ -40,7 +40,7 @@ package("protobuf-cpp")
             package:add("deps", "zlib")
         end
         if package:version():ge("22.0") then
-            package:add("deps", "abseil", {configs = {cxx_standard = (package:is_plat("linux") and "17" or "14")}})
+            package:add("deps", "abseil", {configs = {cxx_standard = "17"}})
```
Comment (SirLynix) [inline packages/p/protobuf-cpp/xmake.lua:43]: same

File: packages/r/re2/xmake.lua [hunk 3]
```diff
@@ -14,11 +14,14 @@ package("re2")
...
     add_versions("2024.04.01", "3f6690c3393a613c3a0b566309cf04dc381d61470079b653afc47c67fb898198")
 
-    add_deps("cmake", "abseil")
+    add_deps("cmake")
 
     if is_plat("linux") then
         add_syslinks("pthread")
     end
+    on_load(function (package)
+        package:add("deps", "abseil", {configs = {cxx_standard = "17"}})
```
Comment (SirLynix) [inline packages/r/re2/xmake.lua:23]: same

File: packages/p/protobuf-cpp/xmake.lua (hunk 2 above)
Comment (MarcusMogg) [reply packages/p/protobuf-cpp/xmake.lua]: fix other deps, but keep here for the  same coding style.

**Pull request merged**